include/boost/geometry/algorithms/detail/distance/segment_to_box.hpp

```diff
@@ -380,24 +379,37 @@ class segment_to_box_2D
                     // segment & crosses band (TODO:merge with box-box dist)
                     if (math::equals(geometry::get<0>(p0), geometry::get<0>(p1)))
                     {
-                        SegmentPoint high = geometry::get<1>(p1) > geometry::get<1>(p0) ? p1 : p0;
-                        if (less_equal(geometry::get<1>(high), geometry::get<1>(top_right)))
+                        SegmentPoint high = geometry::get<1>(p1)
+                                            > geometry::get<1>(p0) ? p1 : p0;
+                        if (less_equal(geometry::get<1>(high),
+                                       geometry::get<1>(top_right)))
                         {
-                            return cast::apply(ps_strategy.apply(high, bottom_right, top_right));
+                            return cast_to_result<ReturnType>::apply(ps_strategy
+                                         .apply(high, bottom_right, top_right));
                         }
-                        return cast::apply(ps_strategy.apply(top_right, p0, p1));
+                        ReturnType res = cast_to_result<ReturnType>::apply
+                                         (ps_strategy.apply(top_right, p0, p1));
+                        strategy::distance::services::swap<SBStrategy>::apply(res);
+                        return res;
                     }
-                    return cast::apply(ps_strategy.apply(p0, bottom_right, top_right));
+                    return cast_to_result<ReturnType>::apply
+                               (ps_strategy.apply(p0, bottom_right, top_right));
                 }
                 // distance is realized between the top-right
                 // corner of the box and the segment
-                return cast::apply(ps_strategy.apply(top_right, p0, p1));
+                ReturnType res = cast_to_result<ReturnType>
+                                  ::apply(ps_strategy.apply(top_right, p0, p1));
```

Comment:
Here the code is formatted in yet another way. :)

Reply:
changed to match the above